services/core/java/com/android/server/wm/ActivityClientController.java

```diff
@@ -705,6 +705,22 @@ public String getLaunchedFromPackage(IBinder token) {
         return null;
     }
 
+    @Override
+    public boolean doesLaunchedFromPackageHavePermission(IBinder token, String permission) {
+        if (getLaunchedFromPackage(token) == null && !WindowManagerHooks.canAccessLaunchedFromPackagePermission()) {
```

Comment:
Are apps that can access getLaunchedFromPackage() allowed to check for caller permission without this patch?

If no, then this patch would grant them additional access.